### PR TITLE
feat(ngrx-entity): convert cases, context to entity adapter

### PR DESCRIPTION
**Pull request by yairtawil**

### CODE REVIEW

File: src/app/packages/menu-items/layers-manager/effects/layers.effects.ts [hunk 1]
```diff
@@ -13,6 +13,7 @@ import { DataLayersService, LayerRootsBundle } from '../services/data-layers.ser
 import 'rxjs/add/operator/map';
 import 'rxjs/add/operator/catch';
 import 'rxjs/add/operator/switchMap';
+import 'rxjs/add/operator/mergeMap';
```
Comment (liorze) [inline src/app/packages/menu-items/layers-manager/effects/layers.effects.ts]: nit: unused import

File: src/app/packages/menu-items/cases/services/cases.service.ts [hunk 2]
```diff
@@ -9,14 +9,18 @@ import { Case } from '../models/case.model';
...
 import { UUID } from "angular2-uuid";
 
 export const casesConfig: InjectionToken<ICasesConfig> = new InjectionToken('cases-config');
 
 @Injectable()
 export class CasesService {
-	static defaultCase: Case;
+	static defaultTime: CaseTimeState = {
+		type: 'absolute',
+		from: new Date(new Date().getTime() - 365 * 24 * 60 * 60 * 1000),
```
Comment (liorze) [inline src/app/packages/menu-items/cases/services/cases.service.ts]: Consider using `moment` for clarity (and readability).

File: confd/production.tmpl [hunk 3]
```diff
@@ -162,17 +162,8 @@
...
-			"bucket": "context",
-			"auth": "elastic:changeme"
-		},
-		"proxy": {
-			"uri": "{{ getv "/ansyn/context/db/proxy/url" "NO-VALUE" }}",
-			"bucket": "contexts",
-			"apiObject" : "Http"
-		}
-    },
+    	"baseUrl": "{{ getv "/ansyn/layers/baseurl" "NO-VALUE" }}"
```
Comment (liorze) [inline confd/production.tmpl]: Maybe we should merge all those env variables into one? We have a key for cases and layers.

File: src/app/app/app-effects/effects/menu.app.effects.spec.ts [hunk 4]
```diff
@@ -1,5 +1,5 @@
 import { AddMenuItemAction, SelectMenuItemAction } from '@ansyn/menu';
-import { AddCaseSuccessAction, CasesReducer, SelectCaseByIdAction } from '@ansyn/menu-items/cases';
+import { AddCaseAction, CasesReducer, SelectCaseByIdAction } from '@ansyn/menu-items/cases';
```
Comment (liorze) [inline src/app/app/app-effects/effects/menu.app.effects.spec.ts]: Unused import

File: src/app/packages/menu-items/cases/effects/cases.effects.ts [hunk 5]
```diff
@@ -79,47 +73,37 @@ export class CasesEffects {
...
-	 */
-	@Effect()
-	onDeleteCaseBackend$: Observable<any> = this.actions$
-		.ofType<DeleteCaseBackendAction>(CasesActionTypes.DELETE_CASE_BACKEND)
-		.map(({ payload }) => payload)
-		.switchMap((deletedCaseId) => {
-			return this.casesService.removeCase(deletedCaseId)
-				.map(() => new DeleteCaseBackendSuccessAction(deletedCaseId));
-		}).share();
+	// /**
```
Comment (liorze) [inline src/app/packages/menu-items/cases/effects/cases.effects.ts]: Did you intend to delete this effect?

File: src/app/packages/menu-items/cases/reducers/cases.reducer.ts [hunk 6]
```diff
@@ -2,128 +2,81 @@ import { CasesActions, CasesActionTypes } from '../actions/cases.actions';
...
-	contexts: Context[];
-	contextsLoaded: boolean;
-	updatingBackend: boolean;
 }
 
-export const initialCasesState: ICasesState = {
-	cases: [],
+export const casesFeatureKey = 'cases';
+
+export function sortByCreationTime(ob1: Case, ob2: Case): number {
```
Comment (liorze) [inline src/app/packages/menu-items/cases/reducers/cases.reducer.ts]: Replace with: `return ob2.creationTime - ob1.creationTime;`.

File: src/app/packages/menu-items/layers-manager/effects/layers.effects.ts (hunk 1 above)
Comment (yairtawil) [reply src/app/packages/menu-items/layers-manager/effects/layers.effects.ts]: not

**Pull request merged**